### PR TITLE
webpack + OffscreenCanvas + mathjs

### PR DESCRIPTION
**Pull request by 2xAA**
This preliminary PR combo fixes a bunch, but may also currently break a few things.
It'll need a thorough review and changes before merge.

### Fixes:
* #3 - mathjs is way overkill
  * Removes `vendor/mathjs.js` in favour of `mathjs-expression-parser` installed via npm
* #20 - Add UMD/global/worker scope build
  * Builds the library to work in any JavaScript scope (specifically Workers)
  * I think currently `/dist/build.js` and `/dist/build-worker.js` actually build to be the same file - probably can just ship the one build file
  * Changes the build process to use webpack for easy build config
* #16 - OffscreenCanvas support
  * Allows OffscreenCanvas to be used as a texture

### Changes:
* Adds Offscreen example at `npm run examples-offscreen`
* Adds .DS_Store to .gitignore
* Updates `/src` to use ES6 syntax
* Updates `/examples` to use ES6 & 7 syntax
* Updates existing example to work with new autoplay policies
* Changes `/build` to `/dist`
  * this can be reverted, just a stylistic thing
* Removes browserify
* Removes `/lib` folder
  * If we need this back let me know. Maybe we can build out to CommonJS using webpack
* Updates tests to use built worker file and tidies code to match existing codebase

### Notes:

~~I've not checked to see if tests are still working - did all of this in the small amount of free time I had on a train 😬 ~~
Tests have been updated and are passing.

Currently babel isn't being used - we can create an ES5 compatible build if necessary, but I'd say most users won't require this considering the use-cases of this library?

### CODE REVIEW

**Comment by msfeldstein** [comment]
This generally seems good, let me know when you feel ready with it and i'll try acouple things out and merge it

**Comment by 2xAA** [comment]
@msfeldstein I've updated the tests and they're passing. Using the built worker file to test with because of testing within Node.

We can use the source if necessary though - will just need a little time to add in babel for transformation.

Would also like to use Puppeteer to test in-browser at some point, but I'll not start that in this PR.

File: examples/app.js [hunk 1]
```diff
@@ -1,73 +1,96 @@
...
-
-var video = null;
-var time = 0;
-function loadFile(src, callback) {
-  fetch("examples/" + src).then(function(response) {
-    response.text().then(function(body) {
-      callback(body);
-    })
-  })
+const ISFRenderer = require('../dist/build.js').Renderer;
```
Comment (msfeldstein) [inline examples/app.js]: why require the built version vs the src version?

Comment (2xAA) [reply examples/app.js]: @msfeldstein I suppose we could use babel-node?

**Comment by msfeldstein** [comment]
this seems great thanks for all your work on this

**Pull request merged**